### ISSUE TITLE
Add support for photo posts

### ISSUE THREAD

**Issue by dshanske** [swentel/indigenous-android#87]

**Comment by swentel** [comment]
Is there a page on indieweb.org that describes what to send exactly ?

**Comment by EdwardHinkle** [comment]
Here is some info: https://indieweb.org/Micropub#Adding_Files

Besides that, you would probably want to support uploading to a Media Endpoint (if you don't already) https://www.w3.org/TR/micropub/#media-endpoint

Happy to provide any more info you need after those two sections :)

**Comment by swentel** [comment]
Thanks for the info!

I don't have a media endpoint support yet, open at #65 - I don't have that on the Drupal side of things either, so it's waiting on that first.

**Comment by EdwardHinkle** [comment]
Completely understandable. I have things in Indigenous for iOS that are being held up because I don't have it in my site yet and have no way to test/use it :)

**Comment by Ruxton** [comment]
At the most basic level Photos are just notes in which the image is considered the main thing... so, while it'd be great to support the media upload endpoint, there's really nothing standing in the way of the PhotoActivity being just like the NoteActivity except with type Photo.

```java
package com.indieweb.indigenous.micropub.post;

import android.os.Bundle;
import android.view.MenuItem;
import com.indieweb.indigenous.R;

public class PhotoActivity extends BasePostActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        canAddImage = true;
        postType = "Photo";
        addCounter = true;
        setContentView(R.layout.activity_note);
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onPostButtonClick(MenuItem item) {
        sendBasePost(item);
    }

}
```

I built said activity, compiled the app and posted https://ignite.digitalignition.net/2018/08/22/98005/ with this activity

**Comment by grantcodes** [comment]
Once this support is added it would be really nice to have an option to just upload to the media endpoint and not create a post, but provide the returned url of the photo. It's a great way to share screenshots and stuff from your own site.

**Comment by swentel** [comment]
Reviewing again, the photo activity code is the same as note, only the success message is different, so I'm not sure yet if it warrants a separate activity right now ? Granted, the way I post pictures right now to my site is by using the 'article' post type, with a 'to-gallery' tag. I use that tag to move it to a different content type on the Drupal side.

If there's no explicit photo post type, I'd rather focus on the media endpoint for now too, but I need to start writing support for that first on the Drupal module as I can't test it :)

**Comment by Ruxton** [comment]
@swentel there is an explicit photo post kind, it has content and a photo - https://indieweb.org/photo

**Comment by swentel** [comment]
reopening, even though I probably won't really use it, I'll still add it.

**Comment by swentel** [comment]
Actually, leaving as is.
